lib/ts/utils.ts

```diff
@@ -274,7 +274,118 @@ export class Deferred<T> {
         });
     }
 
-    attach(prom: Promise<T>) {
+    attach(prom: Promise<T>): void {
         prom.then(this.resolve, this.reject);
     }
 }
+
+export function mergeObjects<T>(obj1: T, obj2: T): T {
+    const res = {
+        ...obj1,
+    };
+    for (const key in obj2) {
+        if (typeof res[key] === "object" && typeof obj2[key] === "object") {
+            res[key] = mergeObjects(res[key], obj2[key]);
+        } else {
+            res[key] = obj2[key];
+        }
+    }
+
+    return res;
+}
+
+export function normaliseCookieScopeOrThrowError(cookieScope: string): string {
```

Comment:
I think you had to reuse this from the supertokens-website lib

Reply:
I moved this (and all other TODOs in the code) into the list in the PR description. In short, I don't think we should because:
- We'd need to rename it in the website repo (right now, it's `normaliseSessionScopeOrThrowError`)
- Submodule imports like this (going into the utils of another package) are not nice
- It's relatively short and really stable code, so we are unlikely to have any trouble with updating it

(the first one is something I didn't realize the last time we spoke about this)

Reply:
Fair enough. Another idea to consider is moving this to the same repo in which you have the JS override stuff?